ansible-scylla-node/tasks/Debian.yml

```diff
@@ -74,15 +69,91 @@
     when: scylla_version == 'latest' and scylla_edition == 'enterprise'
 
   - name: Install specified OSS Scylla
-    apt:
-      name: "scylla={{ scylla_version }}*"
-      state: present
+    block:
+      - name: Get dependencies of scylla package
+        shell: apt depends scylla={{ scylla_version }}*
+        register: aptdependencies
+      - name: Set dependencies of scylla package as fact
+        set_fact:
+          scylladependencies: "{{ aptdependencies.stdout | regex_findall(regexp) }}"
+        vars:
+            regexp: ': ([a-z0-9+-\.]+) \('
+
+      - name: Get dependencies of scylla-tools package
+        shell: apt depends scylla-tools={{ scylla_version }}*
+        register: apttoolsdependencies
+      - name: Set dependencies of scylla package as fact
+        set_fact:
+          scyllatoolsdependencies: "{{ apttoolsdependencies.stdout | regex_findall(regexp) }}"
+        vars:
+            regexp: ': ([a-z0-9+-\.]+) \('
+
+      - name: Get dependencies of scylla-server package
+        shell: apt depends scylla-server={{ scylla_version }}*
+        register: aptserverdependencies
+      - name: Set dependencies of scylla package as fact
+        set_fact:
+          scyllaserverdependencies: "{{ aptserverdependencies.stdout | regex_findall(regexp) }}"
+        vars:
+            regexp: ': ([a-z0-9+-\.]+) \('
+
+      - name: Install specified OSS Scylla packages
+        apt:
+          name: "{{ item }}={{ scylla_version }}*"
```

Comment:
This is a wrong logic. You assume that the version of the package is going to be the same as the version of the corresponding meta package and this is a wrong assumption. 
You should have picked the version of each package from the output of `apt depends`:

```
$ apt depends scylla-enterprise=2021.1.9-*
scylla-enterprise
  Depends: scylla-enterprise-server (= 2021.1.9-0.20220208.86e3ea4df-1)
  Depends: scylla-enterprise-jmx (= 2021.1.9-0.20220208.86e3ea4df-1)
  Depends: scylla-enterprise-tools (= 2021.1.9-0.20220208.86e3ea4df-1)
  Depends: scylla-enterprise-tools-core (= 2021.1.9-0.20220208.86e3ea4df-1)
  Depends: scylla-enterprise-kernel-conf (= 2021.1.9-0.20220208.86e3ea4df-1)
```

And you should assume that each package can have a different version.